packages/zoe/test/unitTests/contracts/loan/test-loan-e2e.js

```diff
@@ -0,0 +1,59 @@
+// ts-check
+import '../../../../exported';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import '@agoric/install-ses';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import test from 'ava';
+import { E } from '@agoric/eventual-send';
+
+import { setupLoanEndToEnd, checkDetails, checkPayout } from './helpers';
+
+test.todo('loan - no mmr');
+test.todo('loan - bad mmr');
+test.todo('loan - no priceOracle');
+test.todo('loan - badPriceOracle');
+test.todo('loan - bad autoswap, no autoswap');
+test.todo('loan - wrong keywords');
+
+test.todo('loan - lend - wrong exit rule');
+test.todo('loan - lend - must want nothing');
```

Comment:
* reschedule liquidation
 * insufficient collateral
 * close before borrow

Reply:
Can't close before borrow  - the method to get the invitation only exists on the borrow offerResult.

Reply:
Also, "borrow not enough collateral" is already a test

Reply:
I thought I saw exit conditions that allowed the lender to close before borrowing took place?

Reply:
The lender can withdraw their assets, but not close the contract. There's a test for that already.

Reply:
These are all tested now